app/views/questions/new.html.erb

```diff
@@ -5,23 +5,74 @@
         <div class="box">
           <h2 class="title-no-margin center">New Question</h2>
           <br>
-            <%= form_for @question do |f| %>
-              <div class="form-group">
-                <%= f.text_field :text, placeholder: 'Question', class: 'form-control', autofocus: true %>
-              </div>
-              <div class="form-group">
-                <%= f.label :qset_id, 'Qset', {class: 'control-label'} %>
-                <%= f.collection_select(:qset_id, Qset.find(current_user.org).descendants, :id, :name, {}, {class: 'form-control'}) %>
+            <%= form_for(@question) do |f| %>
+              <%= f.text_field :text, placeholder: 'Question',  class: 'form-control', autofocus: true %>
+              <button type="button" class="btn btn-primary btn-theme btn-generate-q" data-toggle="modal" data-target="#myModal">
+                Choose Qset
+              </button>
+              <span class="qset-display">
+                <% if !params[:qset].nil? %>
+                  <%= @qsets.find(params[:qset]).name %>
+                <% end %>
+              </span>
+              <!-- hidden input field to pass in qset_id into params on submit -->
+              <% if !params[:qset].nil? %>
+                <input type="text" name="question[qset_id]" id="qset_id_input" class="hide-display" value="<%= @qsets.find(params[:qset]).id %>">
+              <% else %>
+                <input type="text" name="question[qset_id]" id="qset_id_input" class="hide-display">
+              <% end %>
+              <!-- Modal -->
+              <div class="modal fade" id="myModal" tabindex="-1" role="dialog" aria-labelledby="myModalLabel">
+                <div class="modal-dialog" role="document">
+                  <div class="modal-content">
+                    <div class="modal-header">
+                      <button type="button" class="close" data-dismiss="modal" aria-label="Close"><span aria-hidden="true">&times;</span></button>
+                      <h4 class="modal-title" id="myModalLabel">Choose Qset</h4>
+                    </div>
+                    <div class="modal-body">
+                      <!-- displays qsets in hierarchical order -->
+                      <div class="list-group">
+                        <% @qsets.walk_tree do |qset, level| %>
+                          <% if level > 0 %>
+                            <% level_viz = '&nbsp;&nbsp;&nbsp;&nbsp;' * (level - 1) %>
```

Comment:
Maybe `level_indentation`? Lots of this code to create html might be nice if abstracted into a helper (http://mixandgo.com/blog/the-beginner-s-guide-to-rails-helpers) but that isn't necessary for this round.